crates/bevy_render/src/primitives/mod.rs

```diff
@@ -95,6 +95,35 @@ impl Aabb {
     pub fn max(&self) -> Vec3A {
         self.center + self.half_extents
     }
+
+    /// Check if the point is at the front side of the plane.
+    /// Referenced from: [AABB Plane intersection](https://gdbooks.gitbooks.io/3dcollisions/content/Chapter2/static_aabb_plane.html)
+    #[inline]
+    pub fn is_forward_plane(
+        &self,
+        p_normal: &Vec3A,
+        p_distance: f32,
+        world_from_local: &Affine3A,
+    ) -> bool {
+        let aabb_center_world = world_from_local.transform_point3a(self.center);
+        let bb_axis_x_world = world_from_local.x_axis * self.half_extents.x;
+        let bb_axis_y_world = world_from_local.y_axis * self.half_extents.y;
+        let bb_axis_z_world = world_from_local.z_axis * self.half_extents.z;
+        let half_extents_x_world = bb_axis_x_world.dot(Vec3A::X).abs()
+            + bb_axis_y_world.dot(Vec3A::X).abs()
+            + bb_axis_z_world.dot(Vec3A::X).abs();
+        let half_extents_y_world = bb_axis_x_world.dot(Vec3A::Y).abs()
+            + bb_axis_y_world.dot(Vec3A::Y).abs()
+            + bb_axis_z_world.dot(Vec3A::Y).abs();
+        let half_extents_z_world = bb_axis_x_world.dot(Vec3A::Z).abs()
+            + bb_axis_y_world.dot(Vec3A::Z).abs()
+            + bb_axis_z_world.dot(Vec3A::Z).abs();
```

Comment:
All these mutliplications by scalar vectors could be reduced to choosing a single component of a vector.
```
        let bb_axis_x_world = world_from_local.x_axis * self.half_extents.x;
        let bb_axis_y_world = world_from_local.y_axis * self.half_extents.y;
        let bb_axis_z_world = world_from_local.z_axis * self.half_extents.z;
        let half_extents_x_world = bb_axis_x_world.x.abs()
            + bb_axis_y_world.x.abs()
            + bb_axis_z_world.x.abs();
        let half_extents_y_world = bb_axis_x_world.y.abs()
            + bb_axis_y_world.y.abs()
            + bb_axis_z_world.y.abs();
        let half_extents_z_world = bb_axis_x_world.z.abs()
            + bb_axis_y_world.z.abs()
            + bb_axis_z_world.z.abs();
```
Then we can push the `.abs()` all the way back to the vectors (in glam (and by extent bevy_math) you can call .abs() on a vector to apply absolute value to each component)
```
        let bb_axis_x_world_abs = (world_from_local.x_axis * self.half_extents.x).abs();
        let bb_axis_y_world_abs = (world_from_local.y_axis * self.half_extents.y).abs();
        let bb_axis_z_world_abs = (world_from_local.z_axis * self.half_extents.z).abs();
        let half_extents_x_world = bb_axis_x_world_abs.x
            + bb_axis_y_world_abs.x
            + bb_axis_z_world_abs.x;
        let half_extents_y_world = bb_axis_x_world_abs.y
            + bb_axis_y_world_abs.y
            + bb_axis_z_world_abs.y;
        let half_extents_z_world = bb_axis_x_world_abs.z
            + bb_axis_y_world_abs.z
            + bb_axis_z_world_abs.z;
```
We can also notice, we always sum up the same components, this reduces to vector addition
```
        let bb_axis_x_world_abs = (world_from_local.x_axis * self.half_extents.x).abs();
        let bb_axis_y_world_abs = (world_from_local.y_axis * self.half_extents.y).abs();
        let bb_axis_z_world_abs = (world_from_local.z_axis * self.half_extents.z).abs();
        let half_extents_world = bb_axis_x_world_abs + bb_axis_y_world_abs + bb_axis_z_world_abs;
```
Then we note that the absolute value can distribute over the multiplication (there are no additions taking place on that line).
```
        let bb_axis_x_world_abs = world_from_local.x_axis.abs() * self.half_extents.abs().x;
        let bb_axis_y_world_abs = world_from_local.y_axis.abs() * self.half_extents.abs().y;
        let bb_axis_z_world_abs = world_from_local.z_axis.abs() * self.half_extents.abs().z;
        let half_extents_world = bb_axis_x_world_abs + bb_axis_y_world_abs + bb_axis_z_world_abs;
```
Finally, one has to notice that we are effectively just doing matrix - vector multiplication, written out.
Affine3A::matrix3 gives you a matrix containing exactly the columns you access with x_axis, y_axis, z_axis, we can apply absolute value to the entire matrix:
```suggestion
        let half_extents_world = world_from_local.matrix3.abs() * self.half_extents.abs();
```